ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/pull-queries-with-range-scan.json

```diff
@@ -0,0 +1,354 @@
+      },
+      "inputs": [
+        {"topic": "test_topic", "timestamp": 12345, "key": "10", "value": {"v0": "v10a", "v1": "v10b"}},
+        {"topic": "test_topic", "timestamp": 12355, "key": "11", "value": {"v0": "v11a", "v1": "v11b"}},
+        {"topic": "test_topic", "timestamp": 12365, "key": "12", "value": {"v0": "v12a", "v1": "v12b"}},
+        {"topic": "test_topic", "timestamp": 12375, "key": "13", "value": {"v0": "v13a", "v1": "v13b"}},
+        {"topic": "test_topic", "timestamp": 12385, "key": "14", "value": {"v0": "v14a", "v1": "v14b"}}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `V0` STRING, `V1` STRING"}},
+          {"row":{"columns":["13", "v13a", "v13b"]}},
+          {"row":{"columns":["14", "v14a", "v14b"]}}
+        ]}
+      ]
+    },
+    {
+      "name": "greater-equal",
+      "statements": [
+        "CREATE TABLE INPUT (ID STRING PRIMARY KEY, V0 STRING, V1 STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID >= '12';"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  false,
+        "ksql.query.pull.range.scan.enabled":  true
+      },
+      "inputs": [
+        {"topic": "test_topic", "timestamp": 12345, "key": "10", "value": {"v0": "v10a", "v1": "v10b"}},
+        {"topic": "test_topic", "timestamp": 12355, "key": "11", "value": {"v0": "v11a", "v1": "v11b"}},
+        {"topic": "test_topic", "timestamp": 12365, "key": "12", "value": {"v0": "v12a", "v1": "v12b"}},
+        {"topic": "test_topic", "timestamp": 12375, "key": "13", "value": {"v0": "v13a", "v1": "v13b"}},
+        {"topic": "test_topic", "timestamp": 12385, "key": "14", "value": {"v0": "v14a", "v1": "v14b"}}
+      ],
+      "responses": [
+        {"admin": {"@type": "currentStatus"}},
+        {"admin": {"@type": "currentStatus"}},
+        {"query": [
+          {"header":{"schema":"`ID` STRING KEY, `V0` STRING, `V1` STRING"}},
+          {"row":{"columns":["12", "v12a", "v12b"]}},
+          {"row":{"columns":["13", "v13a", "v13b"]}},
+          {"row":{"columns":["14", "v14a", "v14b"]}}
+        ]}
+      ]
+    },
+    {
+      "name": "range scan on non-strings should fail if table-scan is disabled",
+      "statements": [
+        "CREATE TABLE INPUT (ID INTEGER PRIMARY KEY, V0 STRING, V1 STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE MATVIEW AS SELECT ID, V0, V1 FROM INPUT;",
+        "SELECT ID, V0, V1 FROM MATVIEW WHERE ID >= 12;"
+      ],
+      "properties": {
+        "ksql.query.pull.table.scan.enabled":  false,
+        "ksql.query.pull.range.scan.enabled":  true
+      },
+      "expectedError": {
+        "message": "Query requires table scan to enabled"
```

Comment:
I'm not 100% sure about this suggestion, but in the same spirit as above we might want to specifically call out that range scans only work on string keys at the moment in this error message.

Reply:
There might be a way to make this work. Problem is, non-string range scans are not an error and throw no exception but just fallback to table scans, so it's currently hard to catch. Will think about it in a follow-up.